Remotion/Core/Core/Context/SafeContext.Task.cs

```diff
@@ -0,0 +1,126 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+// We must use an alias as the class name is the same
+using Task_ = System.Threading.Tasks.Task;
```

Comment:
I gather you went with the comment? My reasoning was that while reading the code, I first look at `Task_` and am confused what it means. => only after I look around and find the using that things become clearer. The main reasoning it that `Task_` on it's own doesn't give me much context about what's going on plus the trailing underscore isn't something that's part of our style conventions so there's the extra hurdle to figure out the syntax and meaning. 

That's why I want to give Task and ExecutionContext a meaningful prefix (e.g. `SystemTask` or `System_Task`), making the lines much more self explainatiory.

Reply:
Some of my changes did not make it into the review commits because I did change that but well here we go again.